test/test_testing.py

```diff
@@ -4073,6 +4073,228 @@ def test_add_layer_merge(self, client: _TestingPebbleClient):
+            checks:
+              ready:
+                exec:
+                  command: service nginx ready
+                level: ready
+                period: 30s
+                threshold: 1
+              up:
+                exec:
+                  command: service nginx status
+                level: alive
+                override: merge
+                period: 10s
+                threshold: 5
+            """)
+            == client.get_plan().to_yaml()
+        )
+
+    def test_add_layer_log_targets_combine_override_replace(self, client: _TestingPebbleClient):
+        plan = client.get_plan()
+        assert isinstance(plan, pebble.Plan)
+        assert plan.to_yaml() == '{}\n'
+        client.add_layer(
+            'foo',
+            pebble.Layer("""\
+            log-targets:
+                baz:
+                    override: replace
+                    type: loki
+                    location: https://example.com:3100/loki/api/v1/push
+                    services:
+                        - foo,
+                    labels:
+                        key: value,
+                        key1: value1,
+            """),
+        )
+        plan = client.get_plan()
+        # The YAML should be normalized
+        assert (
+            textwrap.dedent("""\
+            log-targets:
+              baz:
+                labels:
+                  key: value,
+                  key1: value1,
+                location: https://example.com:3100/loki/api/v1/push
+                override: replace
+                services:
+                - foo,
+                type: loki
+            """)
+            == plan.to_yaml()
+        )
+        client.add_layer(
+            'foo',
+            pebble.Layer("""\
+            log-targets:
+                baz:
+                    override: replace
+                    type: loki
+                    location: https://example123.com:3100/loki/api/v1/push
+                    services:
+                        - foo,
+                    labels:
+                        key1: value1,
+            """),
+            combine=True,
+        )
+        plan = client.get_plan()
+        assert (
+            textwrap.dedent("""\
+            log-targets:
+              baz:
+                labels:
+                  key1: value1,
+                location: https://example123.com:3100/loki/api/v1/push
+                override: replace
+                services:
+                - foo,
+                type: loki
+            """)
+            == plan.to_yaml()
+        )
+
+    def test_add_layer_log_targets_combine_override_merge(self, client: _TestingPebbleClient):
+        client.add_layer(
+            'foo',
+            pebble.Layer("""\
+            log-targets:
+                baz:
+                    type: loki
+                    location: https://example123.com:3100/loki/api/v1/push
+                    services:
+                        - foo,
```

Comment:
I don't think there should be trailing commas here (I think YAML is treating them as part of the value).

Reply:
It looks like this isn't resolved yet -- there are still trailing commas on these values.

Reply:
My bad. Commas removed now.